### PR TITLE
Devel/sort fix

### PR DESCRIPTION
**Pull request by ebblake**
* **What does this PR do?**
Fixes easier issues I found along the way towards my eventual goal of resolving #2342. In particular, this fixes a regression documented in #2988.

* **Screenshots (if relevant)**

* **Does this PR meet the acceptance criteria?** (This is just a reminder for you,
  this section can be removed if you fulfill it.)

   - Documentation created/updated (you have to edit
     [doc/manual.xml.head](https://www.github.com/neomutt/neomutt/blob/master/doc/manual.xml.head)
     for that)

   - All builds and tests are passing

   - Added [doxygen code documentation](https://neomutt.org/dev/doxygen)
     [syntax](http://www.stack.nl/~dimitri/doxygen/manual/docblocks.html)

   - Code follows the [style guide](https://neomutt.org/dev/coding-style)

* **What are the relevant issue numbers?**
#2988

### CODE REVIEW

File: sort.c [hunk 1]
```diff
@@ -436,6 +439,8 @@ void mutt_sort_headers(struct Mailbox *m, struct ThreadsContext *threads,
   }
   else
   {
+    OptSortReverse = c_sort_aux & SORT_REVERSE;
```
Comment (ebblake) [inline sort.c:442]: Copy+paste error, that should be c_sort, not c_sort_aux

File: commands.c [hunk 2]
```diff
@@ -830,18 +830,18 @@ void mutt_print_message(struct Mailbox *m, struct EmailList *el)
...
   enum SortType sort = SORT_DATE;
 
   switch (mutt_multi_choice(reverse ?
-                                /* L10N: The highlighted letters must match the "Sort" options */
+                                /* L10N: The highlighted letters must match the "Rev-Sort" options */
                                 _("Rev-Sort "
                                   "(d)ate,(f)rm,(r)ecv,(s)ubj,t(o),(t)hread,(u)"
                                   "nsort,si(z)e,s(c)ore,s(p)am,(l)abel?") :
-                                /* L10N: The highlighted letters must match the "Rev-Sort" options */
+                                /* L10N: The highlighted letters must match the "Sort" options */
```
Comment (flatcap) [inline commands.c:844]: Actually, these were correct.
What they're saying is:
The highlighted letters in the string **below**, should match those in the string **mentioned** in the comment.

`L10N` comments are copied to the `.po` files where the order of the strings may not match the code.

File: mutt_thread.c [hunk 3]
```diff
@@ -679,29 +680,28 @@ static int compare_threads(const void *a, const void *b)
...
-  int i, array_size, sort_top = 0;
+  int i, array_size;
+  bool sort_top = false;
 
   /* we put things into the array backwards to save some cycles,
    * but we want to have to move less stuff around if we're
    * resorting, so we sort backwards and then put them back
    * in reverse order so they're forwards */
-  short c_sort = cs_subset_sort(NeoMutt->sub, "sort");
+  short c_sort = cs_subset_sort(NeoMutt->sub, "sort_aux");
```
Comment (flatcap) [inline mutt_thread.c:698]: Please can you change `c_sort` to `c_sort_aux`

**Review by flatcap — CHANGES_REQUESTED** [review]

File: commands.c (hunk 2 above)
Comment (ebblake) [reply commands.c]: okay, that makes more sense. I've removed this hunk of the change in my latest forced push.

File: mutt_thread.c (hunk 3 above)
Comment (ebblake) [reply mutt_thread.c:698]: updated

**Review by flatcap — COMMENTED** [review]
First three commits merged.  Thanks!

**Comment by ebblake** [comment]
Hmm. Configure on FreeBSD failed to find qsort_s, then used qsort_r with the wrong signature. We'll have to improve auto.def probing

**Comment by flatcap** [comment]
I tweaked the `.cirrus.yml` to get a bit more info: https://cirrus-ci.com/task/5053252086530048
(I don't have access to FreeBSD box / vm)

Snippet from `config.log`
```
Failed: cc -g -O2 -std=c99 -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -std=c99 -g -O2 -std=c99 -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ conftest__.c -o conftest__
ld: error: undefined symbol: qsort_s
>>> referenced by conftest__.c:3
>>>               /tmp/conftest__-acba91.o:(main)
>>> did you mean: qsort_b
>>> defined in: /lib/libc.so.7
cc: error: linker command failed with exit code 1 (use -v to see invocation)
============
The failed code was:
extern void qsort_s(void);
int main(void) {
qsort_s();
return 0;
}
```

Snippet from `config.h`
```
#define HAVE_QSORT_R 1
/* #undef HAVE_QSORT_S */
```

**Comment by ebblake** [comment]
Looks like FreeBSD wants us to do this first:

  #define __STDC_WANT_LIB_EXT1__ 1

**Review by flatcap — APPROVED** [review]
Very nice.  Thanks!

**Pull request merged**